plugin-server/src/cdp/consumers/cdp-cyclotron-plugins-worker.consumer.ts

```diff
@@ -0,0 +1,193 @@
+        return trackedFetch(...args)
+    }
+
+    public async executePluginInvocation(invocation: HogFunctionInvocation): Promise<HogFunctionInvocationResult> {
+        const result: HogFunctionInvocationResult = {
+            invocation,
+            finished: true,
+            capturedPostHogEvents: [],
+            logs: [],
+        }
+
+        const pluginId = invocation.hogFunction.template_id?.startsWith('plugin-')
+            ? invocation.hogFunction.template_id.replace('plugin-', '')
+            : null
+
+        const isTestFunction = invocation.hogFunction.name.includes('[CDP-TEST-HIDDEN]')
+
+        result.logs.push({
+            level: 'debug',
+            timestamp: DateTime.now(),
+            message: `Executing plugin ${pluginId}`,
+        })
+        const plugin = pluginId ? PLUGINS_BY_ID[pluginId] : null
+
+        if (!plugin || !pluginId) {
+            result.error = new Error(`Plugin ${pluginId} not found`)
+            result.logs.push({
+                level: 'error',
+                timestamp: DateTime.now(),
+                message: `Plugin ${pluginId} not found`,
+            })
+            return result
+        }
+
+        const addLog = (level: 'debug' | 'warn' | 'error' | 'info', ...args: any[]) => {
+            result.logs.push({
+                level,
+                timestamp: DateTime.now(),
+                message: sanitizeLogMessage(args),
+            })
+        }
+
+        const logger: LegacyPluginLogger = {
+            debug: (...args: any[]) => addLog('debug', ...args),
+            warn: (...args: any[]) => addLog('warn', ...args),
+            log: (...args: any[]) => addLog('info', ...args),
+            error: (...args: any[]) => addLog('error', ...args),
+        }
+
+        let state = this.pluginState[pluginId]
+
+        const fetch = (...args: Parameters<typeof trackedFetch>): Promise<Response> => {
+            if (isTestFunction) {
+                addLog('info', 'Fetch called but mocked due to test function')
+                return Promise.resolve({
+                    status: 500,
+                    json: () =>
+                        Promise.resolve({
+                            message: 'Test function',
+                        }),
+                } as Response)
+            }
+            return this.fetch(...args)
+        }
+
+        if (!state) {
+            // TODO: Modify fetch to be a silent log if it is a test function...
+            const meta: LegacyPluginMeta = {
+                config: invocation.globals.inputs,
+                global: {},
+                fetch,
+                logger: logger,
+            }
+
+            state = this.pluginState[pluginId] = {
+                setupPromise: plugin.setupPlugin?.(meta) ?? Promise.resolve(),
+                meta,
+                errored: false,
+            }
+        }
+
+        try {
+            await state.setupPromise
```

Comment:
question: If I'm reading it correctly, the plugin setup is called once in the lifetime of the worker process, so if it fails for any reason, the process will always return errors for this plugin's call?

Reply:
Yeah indeed. Thats how it worked before too